tests/elk/elastic/2bm86_elastic_stateful.yaml

```diff
@@ -0,0 +1,194 @@
+                    echo 'Elasticsearch is already running, lets check the node is healthy'
+                    HTTP_CODE=$(http "/" "-w %{http_code}")
+                    RC=$?
+                    if [[ ${RC} -ne 0 ]]; then
+                      echo "curl --output /dev/null -k -XGET -s -w '%{http_code}' \${BASIC_AUTH} http://127.0.0.1:9200/ failed with RC ${RC}"
+                      exit ${RC}
+                    fi
+                    # ready if HTTP code 200, 503 is tolerable if ES version is 6.x
+                    if [[ ${HTTP_CODE} == "200" ]]; then
+                      exit 0
+                    elif [[ ${HTTP_CODE} == "503" && "8" == "6" ]]; then
+                      exit 0
+                    else
+                      echo "curl --output /dev/null -k -XGET -s -w '%{http_code}' \${BASIC_AUTH} http://127.0.0.1:9200/ failed with HTTP code ${HTTP_CODE}"
+                      exit 1
+                    fi
+
+                  else
+                    echo 'Waiting for elasticsearch cluster to become ready (request params: "wait_for_status=green&timeout=1s" )'
+                    if http "/_cluster/health?wait_for_status=green&timeout=5s" "--fail" ; then
+                      touch ${START_FILE}
+                      exit 0
+                    else
+                      echo 'Cluster is not yet ready (request params: "wait_for_status=green&timeout=1s" )'
+                      exit 1
+                    fi
+                  fi
+            failureThreshold: 3
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            successThreshold: 3
+            timeoutSeconds: 5
+          ports:
+            - name: http
+              containerPort: 9200
+            - name: transport
+              containerPort: 9300
+          resources:
+            limits:
+              cpu: 1000m
+              memory: 2Gi
+            requests:
+              cpu: 1000m
+              memory: 2Gi
+          env:
+            - name: node.name
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: cluster.initial_master_nodes
+              value: "elasticsearch-master-0"
+            #- name: discovery.type
+            #  value: single-node
+            - name: discovery.seed_hosts
+              value: "elasticsearch-master-headless"
+            - name: cluster.name
+              value: "elasticsearch"
+            - name: network.host
+              value: "0.0.0.0"
+            - name: ES_JAVA_OPTS
+              value: "-Xmx1g -Xms1g"
+            - name: node.data
+              value: "true"
+            - name: node.ingest
+              value: "true"
+            - name: node.master
+              value: "true"
+            - name: rest.action.multi.allow_explicit_index
+              value: "true"
+            - name: ELASTIC_USERNAME
+              value: elastic
+            - name: ELASTIC_PASSWORD
+              value: aGeWkp56b
```

Comment:
We must change this password now 😔

I assume this file wasn't supposed to be committed?

Reply:
it doesn't really matters since we use HTTP for communication instead of HTTPS

Reply:
removing this file

Reply:
Not using SSL (using `http` instead of `https`) means that if you're on a non-secure network (e.g. a wifi at a cafe with no password) anyone on the local network can eavesdrop and see your password in plaintext. This also makes man-in-the-middle attacks possible (dns-hijacking, or malicious ISP, etc.).

None of these are very likely, but we should protect against them.

Having the password exposed publicly (even after the file was removed) means that bots that monitor these kinds of things now have it, making it meaningless to have a password in the first place. Let's please replace it or not use a password at all.

Reply:
replacing this will take all developers changing their local env variable.
I will change the file in 1pass but everyone should download and source it.
as far as I know some developers use their own passwords.

Reply:
another issue will be replacing the password in our current main Kibana.

Reply:
If you plan to implement SSL in the next few days then fine, but in the meantime we should consider this password to be compromised and the current state of any service that uses this password is that it's unprotected.

If it'll take more than a couple of days, then I think the effort to replace this password is well worth it.